module-system/module-implementations/sov-attester-incentives/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+    Vm: Zkvm,
+    Cond: ValidityCondition,
+    Checker: ValidityConditionChecker<Cond>,
+> {
+    /// The address of the token to be used for bonding.
+    pub bonding_token_address: C::Address,
+    /// The address of the account holding the reward token supply
+    pub reward_token_supply_address: C::Address,
+    /// The minimum bond for an attester.
+    pub minimum_attester_bond: u64,
+    /// The minimum bond for a challenger.
+    pub minimum_challenger_bond: u64,
+    /// A code commitment to be used for verifying proofs
+    pub commitment_to_allowed_challenge_method: Vm::CodeCommitment,
+    /// A list of initial provers and their bonded amount.
+    pub initial_attesters: Vec<(C::Address, u64)>,
+    /// The finality period of the rollup (constant)
+    pub rollup_finality_period: u64,
+    /// The current maximum attested height
+    pub maximum_attested_height: u64,
+    /// The light client finalized height
+    pub light_client_finalized_height: u64,
+    /// The validity condition checker used to check validity conditions
+    pub validity_condition_checker: Checker,
+    /// Phantom data that contains the validity condition
+    phantom_data: PhantomData<Cond>,
+}
+
+/// The information about an attester's unbonding
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug)]
+pub struct UnbondingInfo {
+    /// The height at which an attester is allowed to withdraw their tokens
+    pub unbonding_initiated_height: u64,
+    /// The number of tokens that the attester may withdraw
+    pub amount: u64,
+}
+
+/// A new module:
+/// - Must derive `ModuleInfo`
+/// - Must contain `[address]` field
+/// - Can contain any number of ` #[state]` or `[module]` fields
+#[derive(ModuleInfo)]
+pub struct AttesterIncentives<
+    C: sov_modules_api::Context,
+    Vm: Zkvm,
+    Cond: ValidityCondition,
+    Checker: ValidityConditionChecker<Cond>,
+> {
+    /// Address of the module.
+    #[address]
+    pub address: C::Address,
+
+    /// The amount of time it takes to a light client to be confident
+    /// that an attested state transition won't be challenged. Measured in
+    /// number of blocks.
+    #[state]
+    pub rollup_finality_period: sov_state::StateValue<u64>,
+
+    /// The address of the token used for bonding provers
+    #[state]
+    pub bonding_token_address: sov_state::StateValue<C::Address>,
+
+    /// The address of the account holding the reward token supply
+    /// TODO: maybe mint the token before transfering it? The mint method is private in bank
+    /// so we need a reward address that contains the supply.
+    #[state]
+    pub reward_token_supply_address: sov_state::StateValue<C::Address>,
+
+    /// The code commitment to be used for verifying proofs
+    #[state]
+    pub commitment_to_allowed_challenge_method: sov_state::StateValue<StoredCodeCommitment<Vm>>,
+
+    /// Constant validity condition checker for the module.
+    #[state]
+    pub validity_cond_checker: sov_state::StateValue<Checker>,
```

Comment:
It doesn't look like we ever use this checker. It's initialized, but not read AFAICT. What am I missing?

Reply:
Closing the conversation